tests/test_gbytec.F90

```diff
@@ -172,10 +195,73 @@ program test_gbytec
   call g2_sbytesc(out8, in2, iskip, nbits, nskip, num)
   ! Note that the 32-bit IEEE representation of 1.0 is 3f800000. The
   ! decimal for 3f is 63, the decimal for 80 is 128.
-  if (ichar(out4(1)) .ne. 63 .and. ichar(out4(2)) .ne. 128 .and. ichar(out4(3)) .ne. 0 .and. ichar(out4(4)) .ne. 0) stop 50
-  if (ichar(out8(5)) .ne. 63 .and. ichar(out8(6)) .ne. 128 .and. ichar(out8(7)) .ne. 0 .and. ichar(out8(8)) .ne. 0) stop 50
+  if (ichar(out4(1)) .ne. 63 .and. ichar(out4(2)) .ne. 128 .and. &
+       ichar(out4(3)) .ne. 0 .and. ichar(out4(4)) .ne. 0) stop 120
+  if (ichar(out8(5)) .ne. 63 .and. ichar(out8(6)) .ne. 128 .and. &
+       ichar(out8(7)) .ne. 0 .and. ichar(out8(8)) .ne. 0) stop 130
   ! print '(z2.2)', out4(1)  
 
+  print *, '   testing g2_sbytec() with 64-bit int...'
+  in8(1) = 1
+  do i = 1, 8
+     out8(i) = char(0)
+  end do
+  call g2_sbytec(out8, in8, iskip, 64)
+  do i = 1, 8
+     !print '(z2.2)', out8(i)
+     if (i .lt. 8) then
+        if (ichar(out8(i)) .ne. 0) stop 140
+     else
+        if (ichar(out8(i)) .ne. 1) stop 140
+     endif
+  end do
+
+  print *, '   now unpack into 1 64-bit int with g2_gbytesc()...'
+  in8_1(1) = 0
+  call g2_gbytesc(out8, in8_1, iskip, 64, 0, 1)
+  if (in8_1(1) .ne. in8(1)) stop 150
+
+  print *, '   testing g2_sbytec() with 32-bit int array of size 4...'
+  do i = 1, 4
+     in44(i) = 1
+  end do
+  do i = 1, 16
+     out16(i) = char(0)
+  end do
+  call g2_sbytesc(out16, in44, iskip, 32, 0, 4)
+  do i = 1, 16
+     print '(i3, x, z2.2)', i, out16(i)
+  end do
+  
+  print *, '   now unpack into 4 32-bit ints with g2_gbytesc()...'
+  do i = 1, 4
+     in44_1(i) = 0
+  end do
+  call g2_gbytesc(out16, in44_1, iskip, 32, 0, 4)
+  do i = 1, 4
+     if (in44_1(i) .ne. in44(i)) stop 160
+  end do
+
+  print *, '   testing g2_sbytec() with 64-bit int array of size 4...'
+  do i = 1, 4
+     in84(i) = 1
+  end do
+  do i = 1, 32
+     out32(i) = char(0)
+  end do
+  print *, in84
+  call g2_sbytesc(out32, in84, iskip, 64, 0, 4)
+  do i = 1, 32
+     print '(i3, x, z2.2)', i, out32(i)
+  end do
+
+  print *, '   now unpack into 4 64-bit ints with g2_gbytesc()...'
+  do i = 1, 4
+     in84_1(i) = 0
+  end do
+  call g2_gbytesc(out32, in84_1, iskip, 64, 0, 4)
```

Comment:
Are there values that need to be checked here, or is the idea (for now anyway) to just ensure that the routine doesn't fail?

Reply:
I'm going to submit a follow-on with more testing and fix for the int*8 issue.